roles/edpm_nova/tasks/derive_pci.yml

```diff
@@ -0,0 +1,74 @@
+---
+# Copyright 2024 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: EDPM Node derive_novapci_whitelist Config
+  hosts: "{{ edpm_override_hosts | default('all', true) }}"
+  strategy: linear
+  become: true
+  tags:
+    - nova
+  tasks:
+    - name: Check if edpm node has the sriov nova conf
+      ansible.builtin.stat:
+        path: "{{ edpm_nova_config_dest }}/{{ edpm_nova_default_sriov_conf_file }}"
+      register: sriov_nova_conf_check
+
+    - name: Identify the nova conf file used for derive pci
+      ansible.builtin.debug:
+        msg: |
+              Nova conf file used for pci whitelist is
+              "{{ edpm_nova_config_dest }}/{{ edpm_nova_default_sriov_conf_file }}"
+
+    - name: Derive the sriov pci whitelist addr for nic-partition
+      when:
+        - sriov_nova_conf_check.stat.exists
+        - edpm_nova_sriov_derive_pci_whitelist|bool 
+      - name: Create folder for the derive_whitelist script
+        file:
+          state: directory
+          path: /var/lib/openstack/nova_pci_whitelist_script
```

Comment:
@vcandapp can we use module instead of using script file?

Reply:
For feature parity, will leave this as script for now. There seems to be no need for this script to be run as module outside of nic-partition